src/components/StudioTopbar/StudioTopbar.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState, useEffect } from 'react'
+import { Button, Text, Icon } from '@/shared/components'
+import {
+  StyledTopbarBase,
+  StudioContainer,
+  DrawerButton,
+  MemberInfoContainer,
+  MemberTitleText,
+  MemberTextContainer,
+  ChannelInfoContainer,
+  StyledAvatar,
+  TextContainer,
+  DrawerContainer,
+  NewChannel,
+  NewChannelIconContainer,
+  StyledLink,
+  LogoutButton,
+} from './StudioTopbar.style'
+
+type Channel = {
+  name: string
+  avatar: string
+}
+type ChannelInfoProps = {
+  active?: boolean
+  member: string
+  channel: Channel
+  onClick?: React.MouseEventHandler<HTMLDivElement>
+}
+
+type MemberInfoProps = {
+  memberName: string
+  memberAvatar: string
+}
+
+type NavDrawerProps = {
+  active?: boolean
+  memberName: string
+  memberAvatar: string
+  channels: Channel[]
+  currentChannel: Channel
+  onCurrentChannelChange: (channel: Channel) => void
+  onLogoutClick: () => void
+}
+
+const member = {
+  name: 'Mikael Cowan',
+  avatar: 'https://picsum.photos/300/300',
+  channels: [
+    { name: 'Wild Crypto Fan16', avatar: 'https://picsum.photos/200/300' },
+    { name: 'Mild Crypto Skeptic', avatar: 'https://picsum.photos/200/300' },
+    { name: 'Average Cryptocurrency Enjoyer', avatar: 'https://picsum.photos/200/300' },
+  ],
+}
+
+const StudioTopbar: React.FC = () => {
```

Comment:
I know that's an early stage for this, but member, avatar, and channels should be passed via props.

Reply:
`StudioTopbar` is used in `LayoutWithRouting`, so that would mean that we're going to fetch data in routing component and pass it to top bar. I'm not sure if it's a good idea. I believe that we're going to use hook for getting member and active channels so maybe it's better to use that in top bar component - in that case we don't need to pass it  via props. What do you think?

Reply:
Yes that's right this is very likely going to be a hook so not a prob imo

Reply:
Okay, that makes sense.